src/neural-network.js

```diff
@@ -391,46 +392,54 @@ export default class NeuralNetwork {
 
     this._verifyIsInitialized(data);
 
-    while (status.iterations < this.trainOpts.iterations && status.error > this.trainOpts.errorThresh && Date.now() < endTime) {
-      this._trainingTick(data, status);
+    const runTrainingTick = () => {
```

Comment:
The only concern I can think of in making this a lambda is the overhead in the anonymous closure itself.  Generally I don't care about that but where this in the thick of the training, every iteration will be calling this so maybe we should care.  How would you feel about either putting this logic into the `_trainingTick` method or in it's own explicitly defined method?

Reply:
In the sync training, lets not use a closure.  There really is no idea of a "tick", or a small allotment of "synthetic asynchronous" process.  As @freddyC mentioned, it is fairly heavy and can easily be avoided.

Reply:
coming back to this as a pretty good idea, is there any way we can make it so we aren't relying on a closure for internal training?

Reply:
thinking harder about this, I think it is a non-problem.  Consider:
_If_ we had 1,000 training sets, times 20,000 iterations, then we likely have something to care about for those that use this library, right.  Because 1,000 * 20,000 = 20,000,000 may add a couple seconds or so on modern js, but that isn't what is _dryed_ up here, this is the container for the training.

20,000 function calls isn't even a thing.  I say, this makes the js look good.  Lets roll it on in!

Reply:
I fixed the closure issue you guys had. I just did something along the lines of what @freddyC said and moved the logic into `_trainingTick`, with that function returning whether or not it actually ran a train or not.